packages/api/src/jobs/update_just_read_feed.ts

```diff
@@ -0,0 +1,397 @@
+import { lanaugeToCode } from '../utils/helpers'
+import { logger } from '../utils/logger'
+
+export const UPDATE_JUST_READ_FEED_JOB = 'UPDATE_JUST_READ_FEED_JOB'
+
+export interface UpdateJustReadFeedJobData {
+  userId: string
+}
+
+interface Candidate {
+  id: string
+  title: string
+  url: string
+  type: string
+  thumbnail?: string
+  previewContent?: string
+  languageCode: string
+  author?: string
+  dir: string
+  date: Date
+  topic?: string
+  wordCount: number
+  siteIcon?: string
+  siteName?: string
+  folder?: string
+  score?: number
+  publishedAt?: Date
+  subscription?: {
+    name: string
+    type: string
+  }
+}
+
+interface Item {
+  id: string
+  type: string
+}
+
+interface Section {
+  items: Array<Item>
+  layout: string
+}
+
+const libraryItemToCandidate = (
+  item: LibraryItem,
+  subscriptions: Array<Subscription>
+): Candidate => ({
+  id: item.id,
+  title: item.title,
+  url: item.originalUrl,
+  type: 'library_item',
+  thumbnail: item.thumbnail || undefined,
+  previewContent: item.description || undefined,
+  languageCode: lanaugeToCode(item.itemLanguage || 'English'),
+  author: item.author || undefined,
+  dir: item.directionality || 'ltr',
+  date: item.createdAt,
+  topic: item.topic,
+  wordCount: item.wordCount!,
+  siteName: item.siteName || undefined,
+  siteIcon: item.siteIcon || undefined,
+  folder: item.folder,
+  score: item.score,
+  publishedAt: item.publishedAt || undefined,
+  subscription: subscriptions.find(
+    (subscription) =>
+      subscription.name === item.subscription ||
+      subscription.url === item.subscription
+  ),
+})
+
+const publicItemToCandidate = (item: PublicItem): Candidate => ({
+  id: item.id,
+  title: item.title,
+  url: item.url,
+  type: 'public_item',
+  thumbnail: item.thumbnail,
+  previewContent: item.previewContent,
+  languageCode: item.languageCode || 'en',
+  author: item.author,
+  dir: item.dir || 'ltr',
+  date: item.createdAt,
+  topic: item.topic,
+  wordCount: item.wordCount!,
+  siteIcon: item.siteIcon,
+  publishedAt: item.publishedAt,
+  subscription: {
+    name: item.source.name,
+    type: item.source.type,
+  },
+})
+
+const selectCandidates = async (user: User): Promise<Array<Candidate>> => {
+  const userId = user.id
+  // get last 100 library items saved and not seen by user
+  const libraryItems = await searchLibraryItems(
+    {
+      size: 100,
+      includeContent: false,
+      query: `-is:seen wordsCount:>0`,
+    },
+    userId
+  )
+
+  logger.info(`Found ${libraryItems.length} library items`)
+
+  // get subscriptions for the library items
+  const subscriptionNames = libraryItems
+    .filter((item) => !!item.subscription)
+    .map((item) => item.subscription as string)
+
+  const subscriptions = await findSubscriptionsByNames(
+    userId,
+    subscriptionNames
+  )
+
+  // map library items to candidates and limit to 70
+  const privateCandidates: Array<Candidate> = libraryItems
```

Comment:
I suspect we can tweak these counts a lot.

Reply:
Yeah